### DISCUSSION TITLE
Does it support async? How do I use it with async code?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [rq/rq#1802] [Q&A]
Does it support async? How do I use it with async code?

**Comment by ccrvlh** [comment]
@martinenkoEduard yes and no.

You can send async functions to RQ, and the workers can run them: it will check whether that `func` is a coroutine, and if it is it created a `new_event_loop` and `run_until_complete`. 

However, if you are within a `async` context, it won't work, since you will already have a `loop` running (eg: #1697).
So even though the function that the worker should run can be async, the function that actually enqueues the job must be sync.

This is less than ideal to say the least, and it's something we want to tackle soon.
